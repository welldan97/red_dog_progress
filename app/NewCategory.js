import React, { Component } from 'react';
import PropTypes from 'prop-types';

import noop from 'lodash/fp/noop';

import { Button } from 'reactstrap';

export default class Presentational extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isAddingNewCategory: false,
      newCategoryValue: '',
    };
  }

  handleNewCategory = () => {
    this.setState(state => ({
      ...state,
      isAddingNewCategory: true,
    }));
  }

  handleCreateCategory = () => {
    this.props.onCreateCategory({
      name: this.state.newCategoryValue,
    });
    this.setState(state => ({
      ...state,
      isAddingNewCategory: false,
      newCategoryValue: '',
    }));
  }

  handleChange = event => {
    if (event.persist) {
      event.persist();
    }
    this.setState(state => ({
      ...state,
      newCategoryValue: event.target.value,
    }));
  }

  render() {
    const { isAddingNewCategory, newCategoryValue } = this.state;
    if (isAddingNewCategory) {
      return (
        <div>
          <input
            type="text"
            value={newCategoryValue}
            onChange={this.handleChange}
          />
          <Button
            color="primary"
            className="ml-2"
            onClick={this.handleCreateCategory}
          >
            Create
          </Button>
        </div>
      );
    }
    return (
      <Button color="success" block onClick={this.handleNewCategory}>
        Add another category...
      </Button>
    );
  }
}

Presentational.propTypes = {
  onCreateCategory: PropTypes.func,
};

Presentational.defaultProps = {
  onCreateCategory: noop,
};
