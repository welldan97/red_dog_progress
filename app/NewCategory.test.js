import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import noop from 'lodash/fp/noop';

import NewCategory from './NewCategory';

it('It renders', () => {
  const attributes = {
    onCreateCategory: noop,
  };
  const component = renderer.create(<NewCategory {...attributes} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('It creates new Category', () => {
  const onCreateCategory = jest.fn();
  const attributes = {
    onCreateCategory,
  };
  const component = shallow(<NewCategory {...attributes} />);
  component.simulate('click');
  component.find('input[type="text"]').simulate('change', {
    target: {
      value: 'Cookies addiction',
    },
  });
  expect(toJSON(component)).toMatchSnapshot();

  component.find('Button').simulate('click');
  expect(toJSON(component)).toMatchSnapshot();
  expect(onCreateCategory.mock.calls[0][0].name).toEqual('Cookies addiction');
});
