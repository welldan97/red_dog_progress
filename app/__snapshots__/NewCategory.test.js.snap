// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It creates new Category 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value="Cookies addiction"
  />
  <Button
    className="ml-2"
    color="primary"
    onClick={[Function]}
    tag="button"
  >
    Create
  </Button>
</div>
`;

exports[`It creates new Category 2`] = `
<Button
  block={true}
  color="success"
  onClick={[Function]}
  tag="button"
>
  Add another category...
</Button>
`;

exports[`It renders 1`] = `
<button
  className="btn btn-success btn-block"
  onClick={[Function]}
  type="button"
>
  Add another category...
</button>
`;
